### PR TITLE
Add dynamo test configuration

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #80342

This test configuration runs PyTorch's test suite under torchdynamo.
Once stabilized, we will make this default and remove this particular
CI job.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Review by anijain2305 — APPROVED** [review]

**Comment by malfet** [comment]
Hmm, I don't think any of the changes to .circleci should be necessary.
But also, it would be good to outline perf implications and get a review from somebody to devinfra team before merging it

**Pull request closed without merging**
